7_react/react2/week1/github_user_searcher/src/context/GithubSearchContext.jsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, useState } from "react";
+
+export const GithubSearchContext = createContext();
+
+export const GithubSearchProvider = ({ children }) => {
+  const [query, setQuery] = useState("");
+  const [searchResult, setSearchResult] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const fetchUsers = () => {
+    setIsLoading(true);
+    setError(null);
+
+    fetch(`https://api.github.com/search/users?q=${query}`)
```

Comment:
if the query is empty it will throw: `Error fetching: Failed to fetch users`, please handle this scenario